src/pages/history/components/collapse-content/collapse-content.tsx

```diff
@@ -1,47 +1,133 @@
-import Box from '@mui/material/Box';
-import Typography from '@mui/material/Typography'
+import AceEditor from "react-ace/";
+import "ace-builds/src-noconflict/mode-json";
+import "ace-builds/src-noconflict/theme-clouds_midnight";
+import "ace-builds/src-noconflict/theme-solarized_dark";
+import "ace-builds/src-noconflict/theme-dracula";
 
-import _isEmpty from 'lodash/isEmpty'
+import _isEmpty from "lodash/isEmpty";
 
-import { TState } from 'models/state'
+import { TState } from "models/state";
 
-import * as S from './styles'
+import * as S from "./styles";
+import ContentCopyOutlinedIcon from "@mui/icons-material/ContentCopyOutlined";
+import { IconButton } from "shared/components/icon-button";
 
 type Props = {
   state: TState;
-}
+};
 
 export const CollapseContent: React.FC<Props> = ({ state }) => {
+  function onClickCopyBag() {
+    navigator.clipboard.writeText(JSON.stringify(state.bag, null, "\t"));
+  }
+
+  function onClickCopyResult() {
+    navigator.clipboard.writeText(JSON.stringify(state.result, null, "\t"));
+  }
+
+  function onClickCopyActor() {
+    navigator.clipboard.writeText(JSON.stringify(state.actor_data, null, "\t"));
+  }
+
   if (_isEmpty(state.result)) {
     return (
-      <Box sx={{ margin: 1 }}>
-        <Typography variant="h6" component="p" gutterBottom >Result</Typography>
-        <Typography variant='caption' component='p'>No result data...</Typography>
-      </Box>
-    )
+      <S.BoxTable>
+        <S.TitleTable>Result</S.TitleTable>
+        <S.TextTable variant="caption" component="p">
+          No result data...
+        </S.TextTable>
+      </S.BoxTable>
+    );
   }
 
   return (
-    <Box sx={{ margin: 1 }}>
-      <Typography variant="h6" component="p" gutterBottom>Result</Typography>
-      <S.Table size='small' aria-label='result'>
-        <S.TableHead>
-          <S.TableRow>
-            <S.TableCell>Key</S.TableCell>
-            <S.TableCell>Value</S.TableCell>
-          </S.TableRow>
-        </S.TableHead>
-        <S.TableBody>
-          {
-            Object.keys(state?.result).map((key: any, index) => (
-              <S.TableRow key={index}>
-                <S.TableCell>{key}</S.TableCell>
-                <S.TableCell>{JSON.stringify(state.result[key])}</S.TableCell>
-              </S.TableRow>
-            ))
-          }
-        </S.TableBody>
-      </S.Table>
-    </Box>
-  )
-}
\ No newline at end of file
+    <>
+      <S.BoxTable>
+        <S.TitleTable>Bag</S.TitleTable>
+        <IconButton
+          icon={ContentCopyOutlinedIcon}
+          onClick={onClickCopyBag}
+          tooltip="copiar"
+        ></IconButton>
+        <S.Table aria-label="bag">
+          <S.TableBody>
+            <S.TableRow>
+              <AceEditor
```

Comment:
O component AceEditor pode ser passado pra dentro do styles. Incluir atributos que sejam estáticos.

```suggestion
              <S.Editor value={JSON.stringify(state.bag, null, "\t")} />
```